templates/master/00-master/baremetal/files/baremetal-haproxy.yaml

```diff
@@ -99,9 +99,12 @@ contents:
           mountPath: "/var/run/haproxy"
         livenessProbe:
           initialDelaySeconds: 10
-          httpGet:
-            path: /haproxy_ready
-            port: 50936
+          exec:
+            command:
+              - /bin/bash
+              - -c
+              - |
+                [ ! -e /etc/haproxy/haproxy.cfg ] || /usr/bin/curl -kLfs  -o /dev/null  http://localhost:50936/haproxy_ready
```

Comment:
Also, this doesn't seem right. The container should not be considered "alive" when the config file for it doesn't exist. The liveness probe would correctly have been failing until the config got rendered since the /haproxy_ready endpoint is defined in the config.

The existence or non-existence of haproxy.cfg shouldn't affect the reload command since that's talking to socat, not haproxy. Do we have any idea why the monitor couldn't send reload to socat? That seems to be the underlying problem here.

Reply:
Well, yes and no, until the monitor container rendered (it may takes ~20-40 Sec) the config for the first time the haproxy container doesn't run HAProxy LB and I assume it should be considered as 'alive' in that period.

Anyhow, I assume that adjusting the initial delay of the Liveness probe should be good enough in this case, so I'll revert to my first fix.

Now, regarding the unix socket, I agree that we should understand why the client receives error when the server isn't running, though queuing multiple reload messages may cause another problems  :-)